.github/workflows/test.yml

```diff
@@ -1,24 +1,58 @@
 name: Test
 
 on:
-  - pull_request
-  - push
+  push:
+    branches: ["**"]
+  pull_request:
+    branches: ["**"]
 
 jobs:
   test:
     runs-on: ubuntu-latest
     strategy:
       fail-fast: false
       matrix:
-        ruby:
-          - "2.7"
-          - "3.0"
+        include:
```

Comment:
I don't think you need to make it that complicated. Just provide a Ruby version range and let bundler figure out the latest  supporting/working beaker version. You can generate the matrix dynamically: https://github.com/voxpupuli/beaker/blob/master/.github/workflows/test.yml#L28 (and the v2 branch was created yesterday and already supports Ruby 4).

Reply:
Thanks.  I have reduced it significantly now.